From f1348b137f2252d3149c0d7c3da13dc97e0f1bc2 Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Tue, 25 Jul 2017 11:40:22 +0800
Subject: [PATCH 2207/5242] MLK-16075-12 gpu: imx: dpu: common: Add helpers
 dpu_vproc_has/get_fetcheco_cap()

commit  80fb07bebb9497ac442fd130a223d67d927927bd from
https://source.codeaurora.org/external/imx/linux-imx.git

This patch adds helpers dpu_vproc_has/get_fetcheco_cap() support
so that the users may check if a video processing mask has fetcheco
capability or get the fetcheco capability from the mask.
We currently only support fetcheco0 and fetcheco1.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dpu/dpu-common.c |   12 ++++++++++++
 drivers/gpu/imx/dpu/dpu-prv.h    |    4 ++++
 include/video/dpu.h              |    2 ++
 3 files changed, 18 insertions(+)

diff --git a/drivers/gpu/imx/dpu/dpu-common.c b/drivers/gpu/imx/dpu/dpu-common.c
index 477c256..5ab02e2 100644
--- a/drivers/gpu/imx/dpu/dpu-common.c
+++ b/drivers/gpu/imx/dpu/dpu-common.c
@@ -549,6 +549,12 @@ static inline void dpu_cm_write(struct dpu_soc *dpu, u32 value,
 };
 MODULE_DEVICE_TABLE(of, dpu_dt_ids);
 
+bool dpu_vproc_has_fetcheco_cap(u32 cap_mask)
+{
+	return !!(cap_mask & DPU_VPROC_CAP_FETCHECO);
+}
+EXPORT_SYMBOL_GPL(dpu_vproc_has_fetcheco_cap);
+
 bool dpu_vproc_has_hscale_cap(u32 cap_mask)
 {
 	return !!(cap_mask & DPU_VPROC_CAP_HSCALE);
@@ -561,6 +567,12 @@ bool dpu_vproc_has_vscale_cap(u32 cap_mask)
 }
 EXPORT_SYMBOL_GPL(dpu_vproc_has_vscale_cap);
 
+u32 dpu_vproc_get_fetcheco_cap(u32 cap_mask)
+{
+	return cap_mask & DPU_VPROC_CAP_FETCHECO;
+}
+EXPORT_SYMBOL_GPL(dpu_vproc_get_fetcheco_cap);
+
 u32 dpu_vproc_get_hscale_cap(u32 cap_mask)
 {
 	return cap_mask & DPU_VPROC_CAP_HSCALE;
diff --git a/drivers/gpu/imx/dpu/dpu-prv.h b/drivers/gpu/imx/dpu/dpu-prv.h
index e6ea910..5f22cfb 100644
--- a/drivers/gpu/imx/dpu/dpu-prv.h
+++ b/drivers/gpu/imx/dpu/dpu-prv.h
@@ -139,11 +139,15 @@ enum {
 #define DPU_VPROC_CAP_VSCALER4	BIT(1)
 #define DPU_VPROC_CAP_HSCALER5	BIT(2)
 #define DPU_VPROC_CAP_VSCALER5	BIT(3)
+#define DPU_VPROC_CAP_FETCHECO0	BIT(4)
+#define DPU_VPROC_CAP_FETCHECO1	BIT(5)
 
 #define DPU_VPROC_CAP_HSCALE	(DPU_VPROC_CAP_HSCALER4 | \
 				 DPU_VPROC_CAP_HSCALER5)
 #define DPU_VPROC_CAP_VSCALE	(DPU_VPROC_CAP_VSCALER4 | \
 				 DPU_VPROC_CAP_VSCALER5)
+#define DPU_VPROC_CAP_FETCHECO	(DPU_VPROC_CAP_FETCHECO0 | \
+				 DPU_VPROC_CAP_FETCHECO1)
 
 struct dpu_unit {
 	char *name;
diff --git a/include/video/dpu.h b/include/video/dpu.h
index 1ac4a07..a576bd1 100644
--- a/include/video/dpu.h
+++ b/include/video/dpu.h
@@ -631,9 +631,11 @@ void layerblend_pixengcfg_clken(struct dpu_layerblend *lb,
 struct dpu_hscaler *fetchdecode_get_hscaler(struct dpu_fetchdecode *fd);
 struct dpu_vscaler *fetchdecode_get_vscaler(struct dpu_fetchdecode *fd);
 
+bool dpu_vproc_has_fetcheco_cap(u32 cap_mask);
 bool dpu_vproc_has_hscale_cap(u32 cap_mask);
 bool dpu_vproc_has_vscale_cap(u32 cap_mask);
 
+u32 dpu_vproc_get_fetcheco_cap(u32 cap_mask);
 u32 dpu_vproc_get_hscale_cap(u32 cap_mask);
 u32 dpu_vproc_get_vscale_cap(u32 cap_mask);
 
-- 
1.7.9.5

